test/setup-beam.test.js

```diff
@@ -174,856 +159,847 @@ async function testOTPVersions() {
     'https://repo.hex.pm, https://cdn.jsdelivr.net/hex',
     { multiline: true },
   )
+  const previousRunnerArch = process.env.RUNNER_ARCH
```

Comment:
Configuring or restoring environment and mocks should general happen in `before`/`after` or `beforeEach`/`afterEach` blocks. This helps readers understand that setup/teardown code is being run (in contrast to code that dynamically generates test cases) and ensure that the modifications actually happen at the correct time (afaik the test runner doesn't necessarily execute the tests in the moment they are defined using `it)

Reply:
Good to know, thanks. 👍 For now, I just wanted to have a base building block.